Docker/README.md

```diff
@@ -0,0 +1,150 @@
+```
+
+Under the `authorities` folder will be found the independent configuration
+folder for each authority node (it is reminded that each `authority` is 
+constituted from one `primary` node and several `worker` nodes).
+
+The `key.json` file contains the private `key` for the corresponding node which
+is associated to this node only.
+
+The [parameters.json](authorities/parameters.json) file is shared across all the nodes and contains
+the core parameters for a node.
+
+The [committee.json](authorities/committee.json) file is shared across all the nodes and contains
+the information about the authorities (primary & worker nodes), like the public keys, addresses and
+ports available etc.
+
+It has to be noted that the current docker-compose setup is mounting the [Docker/authorities](authorities)
+folder to the service containers in order to share the folders & files in it. That allow us to experiment/change
+configuration without having the need to rebuild the Docker image.
+
+### Docker-compose configuration
+
+The following environment variables are available to be used for each service on the
+[docker-compose.yml](docker-compose.yml) file configuration:
+* `NODE_TYPE` with values `primary|worker` . Defines the node type to bootstrap
+* `AUTHORITY_ID` with decimal numbers, for current setup available values `0..3`. Defines the
+id of the authority that the node/service corresponds to. Basically this defines which
+configuration to use under the `authorities` folder.
+* `LOG_LEVEL` the level of logging for the node defined as number of `v` parameters (e.x `-vvv`). The following
+levels are defined according to the number of "v"s provided: `0 | 1 => "error", 2 => "warn", 3 => "info", 
+4 => "debug", 5 => "trace"`.
+* `CONSENSUS_DISABLED`, this value disables consensus (`Tusk`) for a primary node and enables the
+`gRPC` server. The value that should be passed is `--consensus-disabled`
+* `WORKER_ID` the id, as integer, for service when it runs as a worker
+* `CLEANUP_DISABLED` , when provided with value `true`, it will disable the clean up of the authority folder
+from the database & log data. This is useful to preserve the state between multiple docker compose runs.
+
+### Troubleshooting
+
+#### 1. Compile Errors when building Docker image
+If come across errors while the Docker image is being build, for example errors like:
+```
+error: could not compile `tonic`
+#9 373.3 
+#9 373.3 Caused by:
+#9 373.4   process didn't exit successfully: `rustc --crate-name tonic --edition=2018
+....
+#9 398.4 The following warnings were emitted during compilation:
+#9 398.4 
+#9 398.4 warning: c++: fatal error: Killed signal terminated program cc1plus
+#9 398.4 warning: compilation terminated.
+```
+
+it is possible that the Docker engine is running out of memory and there is no capacity to properly
+compile the code. In this case please try to increase the available RAM at least to 2GB and retry.
+
+#### 2. Mounts denied or cannot start service errors
```

Comment:
Nit: Another frequent one is that "Docker recently deprecated the devicemapper storage driver. Depending on your configuration, you may have to [migrate to an overlayfs driver](https://docs.docker.com/storage/storagedriver/overlayfs-driver/)."

Reply:
Thanks a lot, I'll add this as well here!